### ISSUE TITLE
Re-wrap/split lines at margin... vs. Align paragraphs

### ISSUE THREAD

**Issue by dimon40001** [kvichans/cuda_ext#195]
I would like to have an option to reformat blocks of text keeping the indentation of the first line using the shortcut.

**Align paragraphs** can not be used, because it takes preconfigure values from the Configure... dialog. And it has to be reconfigured each time for each new indentation.

**Re-wrap/split lines at margin...** provides the necessary behavior, thought each time it opens the modal dialog asking for the Margin size and the comment sign.

Would it be possible to improve the behavior of `Re-wrap/split lines at margin...` to not using the modal dialog window?

Currently I refactored the plugin code a bit to add an additional hard-coded value of 120 characters for the right margin.

![additional menu item](https://gcdnb.pbrd.co/images/qCCZMGZMecmj.png?o=1)

**Comment by Alexey-T** [comment]
120 is arbitrary number. Why not 100, 140?

**Comment by Alexey-T** [comment]
Maybe take the value of current right margin, if it’s less than 200?

**Comment by Alexey-T** [comment]
I added the patch in my fork https://github.com/Alexey-T/cuda_ext
branch 'patch-2'
Pls check it

**Comment by dimon40001** [comment]
> 120 is arbitrary number. Why not 100, 140?

Correct, 120 is arbitrary and was used just as an illustration. The best option would be to have a configuration dialog and the command to apply with currently configured settings. In a similar way as it works with Paragraphs functionality.

**Comment by Alexey-T** [comment]
separate config dialog for CudaExt? not a good idea. sorry. in my fork I did that command uses Cud's option "margin" which you can set to 120 in "Plugins / OpTions Editor". it's limited by 2 limits - low and high. AFAIR 40 and 160.

**Comment by dimon40001** [comment]
> branch 'patch-2'
> Pls check it

Thank you. Formatting logic works well. 
I would suggest to take the "default" value not from the Cuda interface-scope option "margin" but from the `Re-wrap/split lines at margin...` dialog. So the dialog already exists.

**Comment by Alexey-T** [comment]
that would need to add saving/loading of entered value in that dialog... not sure its ok.

**Comment by dimon40001** [comment]
As I understand the margin is displayed also as thin vertical line in the editor.

So if one needs to format several paragraphs with several different margins - he would have to go to the settings dialog each time.

As an alternative this setting could be read from the `Paragraph > Align: Configure...`  dialog. Where 3 options can be configured separately specifically for Paragraph commands:

- Paragraph right margin (this value can also be used for Re-wrap... with defaults)
- Indent of first line
- Indent of other lines

**Comment by Alexey-T** [comment]
> this setting could be read from the Paragraph > Align: Configure... dialog. 

done as suggested!
Updated my fork in 'patch-2' branch
